tests/unit/test_logs.c

```diff
@@ -111,3 +111,54 @@ SENTRY_TEST(formatted_log_messages)
     // Transport should be called once
     TEST_CHECK_INT_EQUAL(called_transport, 1);
 }
+
+static void
+test_param_conversion_helper(const char *format, ...)
+{
+    sentry_value_t attributes = sentry_value_new_object();
+    va_list args;
+    va_start(args, format);
+    int param_count = populate_message_parameters(attributes, format, args);
+    va_end(args);
+
+    // Verify we got the expected number of parameters
+    TEST_CHECK_INT_EQUAL(param_count, 3);
+
+    // Verify the parameters were extracted correctly
+    sentry_value_t param0
+        = sentry_value_get_by_key(attributes, "sentry.message.parameter.0");
+    sentry_value_t param1
+        = sentry_value_get_by_key(attributes, "sentry.message.parameter.1");
+    sentry_value_t param2
+        = sentry_value_get_by_key(attributes, "sentry.message.parameter.2");
+
+    TEST_CHECK(!sentry_value_is_null(param0));
+    TEST_CHECK(!sentry_value_is_null(param1));
+    TEST_CHECK(!sentry_value_is_null(param2));
+
+    // Check the values
+    sentry_value_t value0 = sentry_value_get_by_key(param0, "value");
+    sentry_value_t value1 = sentry_value_get_by_key(param1, "value");
+    sentry_value_t value2 = sentry_value_get_by_key(param2, "value");
+
+    TEST_CHECK_INT_EQUAL(sentry_value_as_int64(value0), 1);
+    TEST_CHECK_INT_EQUAL(sentry_value_as_int64(value1), 2);
+    TEST_CHECK_INT_EQUAL(sentry_value_as_int64(value2), 3);
+
+    sentry_value_decref(attributes);
+}
+
+SENTRY_TEST(logs_param_conversion)
+{
+    // TODO this test shows the current limitation for parsing integers on
+    //  32-bit systems
+    int a = 1, b = 2, c = 3;
+#if defined(__i386__) || defined(_M_IX86) || defined(__arm__)
+    // Currently, on 32-bit platforms, we need to cast to long long since the
+    // parameter conversion expects long long for %d/%i format specifiers
+    test_param_conversion_helper(
+        "%lld %lld %lld", (long long)a, (long long)b, (long long)c);
```

Comment:
Let's get rid of `%lld` usage as much as possible on our side of the logging code, because, while `%lld` correctly matches `long long`, `long long` isn't guaranteed to be only 64-bit (it can be broader). But in our logging pipeline, we can only support 64-bit numeric values (at best).

We would have the same problem as with 32-bit systems, on systems that support larger `long long`. That isn't the case very often, and it's not a problem on the platforms we currently support. Still, the Native SDK is continuously integrated on new platforms, and we should reduce the number of issues as much as possible by relying on a safe subset. This should be (applying will probably introduce wrong formatting):

```suggestion
    test_param_conversion_helper(
        "%" PRId64 " %" PRId64 " %" PRId64, (int64_t)a, (int64_t)b, (int64_t)c);
```